src/System.Private.ServiceModel/tests/Common/Infrastructure/OS.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Infrastructure.Common
+{
+    // Helper class for OSID, an enum created to match
+    // RuntimeInformation.OSDescription
+    public static class OS
+    {
+        // These literals must match those returned from RuntimeInformation.OSDescription.
+        private static string WindowsOsName = "Microsoft Windows"; // netcore50 || win8
+        private static string WindowsPhoneOsName = "Microsoft Windows Phone";    // wpa81
+
+        // These literals must partially match what non-Windows OS's self-report
+        private static string CentOSOsName = "CentOS";
+        private static string DebianOsName = "Debian";
+        private static string FedoraOsName = "Fedora";
+        private static string OpenSUSEOsName = "OpenSUSE";
+        private static string OSXOsName = "Darwin";
+        private static string RHELOsName = "RHEL";
+        private static string UbuntuOsName = "Ubuntu";
+
+        private static OSID _currentOSID = 0;
+        private static string _currentOSDescription;
+
+        private static string CurrentOSDescription
+        {
+            get
+            {
+                if (_currentOSDescription == null)
+                {
+                    _currentOSDescription = RuntimeInformation.OSDescription;
+                }
+
+                return _currentOSDescription;
+            }
+        }
+
+        public static OSID Current
+        {
+            get
+            {
+                if (_currentOSID == 0)
+                {
+                    _currentOSID = DetectCurrentOS();
+                    Console.WriteLine(String.Format("Detected current OS as \"{0}\" from description \"{1}\"",
```

Comment:
Yes, these outputs are intentional, in the same way the Console writes are in other areas like certificate loading.  It captures information in the console logs on lab runs that help debug.  I'd like to leave these in at least for awhile and maybe permanently.